PetAdoptions/petsite/petsite/Controllers/HomeController.cs

```diff
@@ -1,199 +1,165 @@
+
+        //Prometheus metric to count the number of kitten searches performed
+        private static readonly Counter KittenSearchCount =
+            Metrics.CreateCounter("petsite_pet_kitten_searches_total", "Count the number of kitten searches performed");
+
+        //Prometheus metric to count the number of bunny searches performed
+        private static readonly Counter BunnySearchCount =
+            Metrics.CreateCounter("petsite_pet_bunny_searches_total", "Count the number of bunny searches performed");
+
+        private static readonly Gauge PetsWaitingForAdoption = Metrics
+            .CreateGauge("petsite_pets_waiting_for_adoption", "Number of pets waiting for adoption.");
+
+
+
+        public HomeController(ILogger<HomeController> logger, IConfiguration configuration, PetSite.Services.IPetSearchService petSearchService, IHttpClientFactory httpClientFactory)
+        {
+            _configuration = configuration;
+            _petSearchService = petSearchService;
+            _httpClientFactory = httpClientFactory;
+            _logger = logger;
+
+            _variety.PetTypes = new List<SelectListItem>()
+            {
+                new SelectListItem() {Value = "all", Text = "All"},
+                new SelectListItem() {Value = "puppy", Text = "Puppy"},
+                new SelectListItem() {Value = "kitten", Text = "Kitten"},
+                new SelectListItem() {Value = "bunny", Text = "Bunny"}
+            };
+
+            _variety.PetColors = new List<SelectListItem>()
+            {
+                new SelectListItem() {Value = "all", Text = "All"},
+                new SelectListItem() {Value = "brown", Text = "Brown"},
+                new SelectListItem() {Value = "black", Text = "Black"},
+                new SelectListItem() {Value = "white", Text = "White"}
+            };
+        }
+
+
+
+        [HttpGet("housekeeping")]
+        public async Task<IActionResult> HouseKeeping()
+        {
+            EnsureUserId();
+            _logger.LogInformation("In Housekeeping, trying to reset the app.");
+            
+            string cleanupadoptionsurl = SystemsManagerConfigurationProviderWithReloadExtensions.GetConfiguration(_configuration,"CLEANUP_ADOPTIONS_URL");
+            
+            using var httpClient = _httpClientFactory.CreateClient();
+            await httpClient.PostAsync(cleanupadoptionsurl, null);
+
+            return View();
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Index(string selectedPetType, string selectedPetColor, string petid)
+        {
+            EnsureUserId();
+            // Add custom span attributes using Activity API
+            var currentActivity = Activity.Current;
+            if (currentActivity != null)
+            {
+                currentActivity.SetTag("pet.type", selectedPetType);
+                currentActivity.SetTag("pet.color", selectedPetColor);
+                currentActivity.SetTag("pet.id", petid);
+                
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Avoid string interpolation within log messages, as it can blur the lines between variables and the log content. Instead, opt for structured logging, where variables are passed as separate arguments. This improves the clarity of logs and facilitates efficient log analysis. For more information, refer to https://github.com/NLog/NLog/wiki/How-to-use-structured-logging.